dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -62,13 +64,75 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         }
     }
 
+    void PushNotificationManager::RegisterUnpackagedApplicationHelper(
+        winrt::guid remoteId,
+        _Out_ wil::unique_cotaskmem_string &unpackagedAppUserModelId)
+    {
+            THROW_IF_FAILED(::CoInitializeEx(nullptr, COINITBASE_MULTITHREADED));
+
+            auto scopeExit = wil::scope_exit(
+                [&]() { CoUninitialize(); });
+
+            {
+                wil::com_ptr<INotificationsLongRunningPlatform> notificationPlatform{
+                    wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+
+                wchar_t processName[1024];
+                THROW_HR_IF(ERROR_FILE_NOT_FOUND, GetModuleFileNameExW(GetCurrentProcess(), NULL, processName, sizeof(processName) / sizeof(processName[0])) == 0);
+
+                winrt::check_hresult(notificationPlatform->RegisterFullTrustApplication(processName, remoteId, &unpackagedAppUserModelId));
+            }
+
+        return;
+    }
+
+    winrt::hresult PushNotificationManager::CreateChannelWithRemoteIdHelper(
+        winrt::guid remoteId,
+        _Out_ wil::unique_cotaskmem_string& channelUri,
+        _Out_ wil::unique_cotaskmem_string& channelId,
+        _Out_ wil::unique_cotaskmem_string& appUserModelIdString,
+        winrt::Windows::Foundation::DateTime& channelExpiryTime
+        )
+    {
+        wchar_t appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+        UINT32 appUserModelIdSize = ARRAYSIZE(appUserModelId);
+
+        THROW_IF_FAILED(GetCurrentApplicationUserModelId(&appUserModelIdSize, appUserModelId));
+
+        THROW_HR_IF(E_INVALIDARG, (appUserModelIdSize > APPLICATION_USER_MODEL_ID_MAX_LENGTH) || (appUserModelIdSize == 0));
+
+        /* channel request result */
+        HRESULT operationalCode;
+        ABI::Windows::Foundation::DateTime channelexpirytime{};
+
+        HRESULT hr = PushNotifications_CreateChannelWithRemoteIdentifier(
+            appUserModelId,
+            remoteId,
+            &operationalCode,
+            &channelId,
+            &channelUri,
+            &channelexpirytime);
+
+        /*
+           As the channel request frameworkUDK API is introduced recently, it is
+           not applicable for older versions. So for error E_NOTIMPL we fallback
+           to calling into
+           CreatePushNotificationChannelForApplicationAsync
+           to request for channels
+        */
+        THROW_HR_IF(hr, (hr != E_NOTIMPL) && (hr != S_OK));
+        THROW_HR_IF(operationalCode, (hr == S_OK) && (operationalCode != S_OK));
```

Comment:
I would remove these 2 line checks and do it like this:
THROW_HR_IF(hr, FAILED(hr) && hr != E_NOT_IMPL);
THROW_IF_FAILED(operationalCode); // when we reach this line, it is assumed that hr == S_OK